### PR TITLE
Add Multichain Session Enabling Signature

### PR DESCRIPTION
**Pull request by filmakarov**
- Added Multichain Signatures for Enabling the Session. Signature is now over the object that consists of Session Hashes and chainIds. 
- Refactored tests , so they all inherit the SmartSessionTestBase

### CODE REVIEW

File: contracts/lib/MultichainHashLib.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        if (providedHash != computedHash) {
+            revert HashMismatch(providedHash, computedHash);
+        }
+
+        digest = keccak256(enableData.hashesAndChainIds);
+    }
+
+    function _parseHashAndChainIdByIndex(bytes memory hashesAndChainIds, uint8 index) internal pure returns (uint64 chainId, bytes32 hash) {
```
Comment (zeroknots) [inline contracts/lib/MultichainHashLib.sol:34]: should we add this to EncodeLib?

File: test/MultiKeySignerTest.t.sol [hunk 2]
```diff
@@ -146,19 +135,32 @@ contract MultiKeySignerTest is SmartSessionBaseTest {
...
+            hashesAndChainIds: "",
+            sessionToEnable: session,
             permissionEnableSig: ""
         });
 
-        bytes32 hash =
-            smartSession.getDigest(enableData.isigner, instance.account, enableData, SmartSessionMode.UNSAFE_ENABLE);
-        // ERC1271
+        bytes32 sessionDigest = smartSession.getDigest(session.isigner, instance.account, session, SmartSessionMode.UNSAFE_ENABLE);
+        enableData.hashesAndChainIds = abi.encodePacked(
```
Comment (zeroknots) [inline test/MultiKeySignerTest.t.sol:155]: we should also add this to encode lib

File: contracts/lib/MultichainHashLib.sol (hunk 1 above)
Comment (filmakarov) [reply contracts/lib/MultichainHashLib.sol:34]: done

File: test/MultiKeySignerTest.t.sol (hunk 2 above)
Comment (filmakarov) [reply test/MultiKeySignerTest.t.sol]: done

**Pull request merged**